### PR TITLE
Vertical two-up

### PR DESCRIPTION
**Pull request by jakearchibald**
The designs show the two-up working vertically on mobile, so I implemented that.

### CODE REVIEW

File: src/components/Output/custom-els/TwoUp/index.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+const legacyClipCompatAttr = 'legacy-clip-compat';
+const verticalAttr = 'vertical';
 
 /**
  * A split view that the user can adjust. The first child becomes
  * the left-hand side, and the second child becomes the right-hand side.
  */
 export default class TwoUp extends HTMLElement {
-  static get observedAttributes () { return [legacyClipCompat]; }
+  static get observedAttributes() { return [verticalAttr]; }
```
Comment (jakearchibald) [inline src/components/Output/custom-els/TwoUp/index.ts]: I had `legacyClipCompat` here previously, but it was a mistake.

File: src/components/Output/custom-els/TwoUp/index.ts [hunk 2]
```diff
@@ -52,39 +53,64 @@ export default class TwoUp extends HTMLElement {
...
+    });
+  }
+
   /**
    * If true, this element works in browsers that don't support clip-path (Edge).
    * However, this means you'll have to set the height of this element manually.
    */
-  get noClipPathCompat () {
-    return this.hasAttribute(legacyClipCompat);
+  get legacyClipCompat() {
```
Comment (jakearchibald) [inline src/components/Output/custom-els/TwoUp/index.ts]: I think I renamed the attr but forgot to rename the prop.

File: src/components/Output/custom-els/TwoUp/index.ts [hunk 3]
```diff
@@ -52,39 +53,64 @@ export default class TwoUp extends HTMLElement {
...
+      this.removeAttribute(legacyClipCompatAttr);
+    }
+  }
+
+  /**
+   * Split vertically rather than horizontally.
+   */
+  get vertical() {
+    return this.hasAttribute(verticalAttr);
+  }
```
Comment (jakearchibald) [inline src/components/Output/custom-els/TwoUp/index.ts]: There's repetition here, but I'm not sure if there's a typescript-friendly way to solve it.

I'd usually transform the attr names to property names then generate the getters/setters automatically.

File: src/components/Output/custom-els/TwoUp/index.ts [hunk 4]
```diff
@@ -1,14 +1,15 @@
 import * as styles from './styles.css';
 import { PointerTracker, Pointer } from '../../../../lib/PointerTracker';
 
-const legacyClipCompat = 'legacy-clip-compat';
+const legacyClipCompatAttr = 'legacy-clip-compat';
+const verticalAttr = 'vertical';
```
Comment (PaulKinlan) [inline src/components/Output/custom-els/TwoUp/index.ts]: Is this meant to be like the disabled attribute, where by presence indicates the state? I was wondering why this wasn't just an `orientation` attribute with a type that has a default of horizontal, but the option of 'vertical'

File: src/components/Output/custom-els/TwoUp/index.ts (hunk 3 above)
Comment (surma) [reply src/components/Output/custom-els/TwoUp/index.ts]: We could write a decorator for tying attrs with props. Not sure if it’s worth it (yet).

Comment (jakearchibald) [reply src/components/Output/custom-els/TwoUp/index.ts]: If we feel we need it, we should talk to the polymer folks. Pretty sure they'll have thought about this before.

**Comment by jakearchibald** [comment]
@PaulKinlan @surma Ready for further review.

I've changed this to be an `orientation` property/attribute with `"vertical"` and `"horizontal"` values.

Setting `el.orientation` just sets the attribute.

Getting `el.orientation` will always return `"vertical"` or `"horizontal"`. This seems to be how `input.type` behaves.

**Pull request merged**